sdk/src/main/scala/io/horizen/utxo/SidechainNodeViewHolder.scala

```diff
@@ -247,8 +247,44 @@ object SidechainNodeViewHolderRef {
             applicationWallet: ApplicationWallet,
             applicationState: ApplicationState,
             genesisBlock: SidechainBlock): Props =
-    Props(new SidechainNodeViewHolder(sidechainSettings, historyStorage, consensusDataStorage, stateStorage, forgerBoxStorage, utxoMerkleTreeProvider, walletBoxStorage, secretStorage,
-      walletTransactionStorage, forgingBoxesInfoStorage, cswDataProvider, backupStorage, params, timeProvider, applicationWallet, applicationState, genesisBlock)).withMailbox("akka.actor.deployment.prio-mailbox")
+    Props(createNodeViewHolder(sidechainSettings, historyStorage, consensusDataStorage, stateStorage, forgerBoxStorage,
+      utxoMerkleTreeProvider, walletBoxStorage, secretStorage, walletTransactionStorage, forgingBoxesInfoStorage,
+      cswDataProvider, backupStorage, params, timeProvider, applicationWallet, applicationState, genesisBlock)).withMailbox("akka.actor.deployment.prio-mailbox")
+
+  private def createNodeViewHolder(sidechainSettings: SidechainSettings, historyStorage: SidechainHistoryStorage,
+                                   consensusDataStorage: ConsensusDataStorage, stateStorage: SidechainStateStorage,
+                                   forgerBoxStorage: SidechainStateForgerBoxStorage,
+                                   utxoMerkleTreeProvider: SidechainStateUtxoMerkleTreeProvider,
+                                   walletBoxStorage: SidechainWalletBoxStorage, secretStorage: SidechainSecretStorage,
+                                   walletTransactionStorage: SidechainWalletTransactionStorage,
+                                   forgingBoxesInfoStorage: ForgingBoxesInfoStorage,
+                                   cswDataProvider: SidechainWalletCswDataProvider, backupStorage: BackupStorage,
+                                   params: NetworkParams, timeProvider: NetworkTimeProvider,
+                                   applicationWallet: ApplicationWallet, applicationState: ApplicationState,
+                                   genesisBlock: SidechainBlock) = {
+    if (params.isHandlingTransactionsEnabled)
+      new SidechainNodeViewHolder(sidechainSettings, historyStorage, consensusDataStorage, stateStorage, forgerBoxStorage, utxoMerkleTreeProvider, walletBoxStorage, secretStorage,
+      walletTransactionStorage, forgingBoxesInfoStorage, cswDataProvider, backupStorage, params, timeProvider, applicationWallet, applicationState, genesisBlock)
+    else
+      new SidechainNodeViewHolder(sidechainSettings, historyStorage, consensusDataStorage, stateStorage, forgerBoxStorage, utxoMerkleTreeProvider, walletBoxStorage, secretStorage,
+        walletTransactionStorage, forgingBoxesInfoStorage, cswDataProvider, backupStorage, params, timeProvider, applicationWallet, applicationState, genesisBlock){
+        override protected def updateMemPool(removedBlocks: Seq[SidechainBlock], appliedBlocks: Seq[SidechainBlock],
```

Comment:
This looks tricky to me. It looks clean, but for someone who's not aware of this code, they will have a hard time debugging a potential problem, not knowing that the methods were overwritten here.
Also would be tricky to add some new conditions

Do you think having a check in the SidechainNodeViewHolder itself is worse? 
```
  override protected def updateMemPool(removedBlocks: Seq[SidechainBlock], appliedBlocks: Seq[SidechainBlock],
                                       memPool: SidechainMemoryPool, state: SidechainState): SidechainMemoryPool = {
    if (params.isHandlingTransactionsEnabled)
      super.updateMemPool(removedBlocks, appliedBlocks, memPool, state)
    else
      memPool
  }
```
same for applyLocallyGeneratedTransactions

Reply:
This is a good point. For sure having the check in the SidechainNodeViewHolder itself is clearer, I didn't do it because I wanted to avoid repeating a check that cannot change during run-time. What if, instead of using an anonymous class, I use a regular class (e.g.  SidechainNodeViewHolderForSeederNode)? Do you think it would be clearer?

Reply:
Yes, sounds good

Reply:
Ok, I'll try this way then.